Add manual hermes source maps iOS steps (#6552)
getsentry/sentry-docs: src/platforms/react-native/manual-setup/hermes.mdx

## Source Maps

If you do not need custom source maps, the `sentry.gradle` build step fully supports Hermes source maps.

## Custom Source Maps

> To manually generate source maps follow this three-step process. Be aware that it's important to exactly match the steps of your React Native tooling. Otherwise the generated bundle and source maps might be working incorrectly, for example pointing to unrelated lines. The following command are examples of the default flow.

Reviewer: `follow this three-step process.` this reads a bit confusing to me. 
First, a reference to match them exactly with my tooling. Then another header `Compile Source maps` followed by a reference to `not all the steps are required.`?

Reviewer: Could this be a generic list of steps instead?
i.e.
- Do something
- (if not `EXTRA_PACKAGER_ARGS` on `iOS`) do the other thing
- ...
- profit!

Author: I see, I'll try to rephrase it.

The React Native tooling steps are extracted from [`react-native-xcode.sh`](https://github.com/facebook/react-native/blob/0.71-stable/scripts/react-native-xcode.sh) for iOS and from [`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) for Android. 

<Note>

[`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) is only available from `0.71`. Version before use [`react.gradle`](https://github.com/facebook/react-native/blob/0.70-stable/react.gradle).

</Note>

### Compile Source maps

> Depending on how your app bundle was generated not all the steps are required. If you use `EXTRA_PACKAGER_ARGS` on `iOS` only the first step applies. This is the default behavior if you used [`@sentry/wizard`](/platforms/react-native/#install). If you use [`SOURCEMAP_FILE`](https://reactnative.dev/docs/sourcemaps#enable-source-maps-on-ios) all steps are necessary.

Reviewer 2: ```suggestion
Depending on how your app bundle was generated not all these compilation steps are required:

- If you use `EXTRA_PACKAGER_ARGS` on `iOS`, only the first step applies. This is the default behavior if you used [`@sentry/wizard`](/platforms/react-native/#install). 
- If you use [`SOURCEMAP_FILE`](https://reactnative.dev/docs/sourcemaps#enable-source-maps-on-ios), all steps are necessary.
```

> 

Reviewer 2: ```suggestion
- If you use Hermes with JS Bundle only the first step applies, but you must replace `main.jsbundle.packager.map` and `index.android.bundle.packager.map` with `main.jsbundle.map` and `index.android.bundle.map`, respectively, in the following commands.

```

Reviewer 2: Related comment for removing the Note below

Reviewer 2: This also looked confusing to me because `main.jsbundle.packager.map` doesn't appear in the below commands (but `main.jsbundle.map` does) and the reverse is also true, `index.android.bundle.packager.map` DOES appear in the below commands (but `index.android.bundle.map` doesn't). 

So I'm wondering if something is backwards here?

<Note>

> We recommend you to use bundle and source map generated by RN tooling. In case you are having issues with the manually generated artifacts take a look at this [issue](https://github.com/facebook/react-native/issues/34043) for more details.

Reviewer 2: ```suggestion
We recommend using the bundle and source maps generated by React Native tooling. If you are having issues with the manually generated artifacts see [this issue](https://github.com/facebook/react-native/issues/34043) for more information.
```

The React Native tooling steps are extracted from [`react-native-xcode.sh`](https://github.com/facebook/react-native/blob/0.71-stable/scripts/react-native-xcode.sh) for iOS and from [`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) for Android. 

<Note>

[`BundleHermesCTask.kt`](https://github.com/facebook/react-native/blob/0.71-stable/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt) is only available from version `0.71` on. For previous versions, use [`react.gradle`](https://github.com/facebook/react-native/blob/0.70-stable/react.gradle).

</Note>

### Compile Source maps

Depending on how your app bundle was generated not all these compilation steps are required:

- If you use `EXTRA_PACKAGER_ARGS` on `iOS`, only the first step applies. This is the default behavior if you used [`@sentry/wizard`](/platforms/react-native/#install). 
- If you use [`SOURCEMAP_FILE`](https://reactnative.dev/docs/sourcemaps#enable-source-maps-on-ios), all steps are necessary.

> - If you use Hermes with JS Bundle only the first step applies, but you must replace `main.jsbundle.packager.map` and `index.android.bundle.packager.map` with `main.jsbundle.map` and `index.android.bundle.map`, respectively, in the following commands.

Author: ```suggestion
```

npx react-native bundle --platform ios --dev false --entry-file index.js --reset-cache --bundle-output main.jsbundle --sourcemap-output main.jsbundle.map --minify false
```

> <Note>
> 
> If you are running Hermes with JS Bundle you only need step one, but replace `main.jsbundle.packager.map` and `index.android.bundle.packager.map` with `main.jsbundle.map` and `index.android.bundle.map` in the commands respectively.
> 
> </Note>

Reviewer 2: ```suggestion
```

Does it make sense to combine this note with the above info on which steps are required based on your app bundle was generated?
